packages/ia-components/sandbox/AnchorDetails.jsx

```diff
@@ -27,27 +26,31 @@ const debug = require('debug')('ia-components:AnchorDetails');
  *
  */
 
-export default class AnchorDetails extends IAReactComponent {
+export default class AnchorDetails extends React.Component {
   // Component that encapsulates the difference between four options: Dweb||IAUX for links.
 
   /*
     React+!Dweb: no onClick unless want analytics
     React+Dweb:  onClick={this.click}
     */
+
   constructor(props) {
-    super(props); // { identifier, reload }
-    //TODO-STATE this might have the issue of constructor not being re-run and needing componentDidMount catch
-    this.setState({
-      urlProps: ObjectFilter(this.props, (k, v) => AnchorDetails.urlparms.includes(k)),
-      anchorProps: ObjectFilter(this.props, (k, v) => (!AnchorDetails.urlparms.includes(k) && !['children'].includes(k)))
-    });
+    super(props);
+    this.onClick = this.onClick.bind(this);
+    this.state = {};
+  }
+  static getDerivedStateFromProps(props, unusedState) {
```

Comment:
question: With the nature of this lifecycle's constant firing, usually changes are only applied when values have changed - this check must be done manually.  I do not see the manual check here, is this function going to hang IF there are many of these components in a certain view?

Reply:
I'm not sure I understand you ... or how multiple components in review would cause a hang (which suggests I'm missing something about React). 

My understanding was that it was componentDidUpdate that needs the check, because it will fire itself if it changes state while getDerivedStateFromProps only ever fires once on a change. .https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Note - I'm skipping answer on same question on other places below, will leave as placeholder till resolved here ....

Reply:
Hi @mitra42 ,
Pardon me for not being clear.  I hope I can do better here.

I am touching on the other two points mentioned in the doc -
> If you want to re-compute some data only when a prop changes
> If you want to “reset” some state when a prop changes

From what you are returning, it seems like you want to do either one of those two points.  Is that off-base?  If so, do let me know.

But, with that assumption + `getDerivedStateFromProps` event firing with every update (click events, state updates, etc.), it seems that running the filter functions on every return can get costly.  Especially if one stacks multiple components on the page with this same pattern.

What the doc recommends may be a bit overkill for our immediate needs, so I am suggesting to only return new state values when the values change. 
-------
What are your thoughts?

Reply:
The techniques suggested under the `re-compute` and `reset` lines are way complicated, I've never understood either of them. 

You don't have access to previous props in getDerivedStateFromProps, so I can't compare props to see if they have changed.   And once I've compute the new state (filtering props into two buckets), then I don't believe there is any advantage to not returning that state, and there is more computation to do to compare state). Note the operations done are cheap - a filter of the props into two buckets. 

Am I missing something (note I've suggested a call about this, otherwise I think this back and forth with 16 day gaps is going to take forever)

I believe this is the same for the other places I've used this pattern, but I'll recheck them once we get clarity on this case.